src/plugins/explore/public/components/visualizations/style_panel/utils.tsx

```diff
@@ -30,12 +30,34 @@ export const DebouncedTruncateField: React.FC<{
   );
 };
 
+export const DebouncedTruncateGaugeBaseField: React.FC<{
+  value?: number | undefined;
+  onChange: (value: number | undefined) => void;
+  label: string;
+  testId: string;
+}> = ({ value, onChange, label, testId }) => {
+  const [localValue, handleChange] = useDebouncedValue(value, onChange);
+
+  return (
+    <EuiFormRow label={label} helpText="Leave empty to calculate based on all values">
```

Comment:
If this component is supposed to be threshold panel specific, better move it to `style_panel`